### PR TITLE
fix: get_all_referenced_columns includes columns referenced in HAVING

### PR DESCRIPTION
**Pull request by lynnagara**
Include columns referecend in the HAVING clause in
get_all_referenced_columns(). Also adds methods for getting the
conditions and having columns of a query which will be used in
https://github.com/getsentry/snuba/pull/554.

### CODE REVIEW

**Review by fpacifici — COMMENTED** [review]
could you please sentry snuba tests ? This potentially changes the behavior of the query processing (unlikely, but let's be safe on this)

**Comment by lynnagara** [comment]
Well... I ran the Sentry tests locally and though I never managed to get a single run all green, none of those failures I experienced seem relevant to this or are specific to this branch....

File: snuba/query/query.py [hunk 1]
```diff
@@ -217,14 +217,31 @@ def get_all_referenced_columns(self) -> Sequence[Any]:
...
+        col_exprs: MutableSequence[Any] = []
+        self.__add_flat_conditions(col_exprs, self.get_conditions())
+        return self.__get_referenced_columns(col_exprs)
+
+    def get_columns_referenced_in_having(self) -> Sequence[Any]:
+        col_exprs: MutableSequence[Any] = []
+        self.__add_flat_conditions(col_exprs, self.get_having())
+        return self.__get_referenced_columns(col_exprs)
+
+    def __add_flat_conditions(self, col_exprs: MutableSequence[Any], conditions=Optional[Sequence[Condition]]):
```
Comment (fpacifici) [inline snuba/query/query.py]: please add None return type.

**Review by fpacifici — APPROVED** [review]

**Review by fpacifici — COMMENTED** [review]
Also please fix the docstring of get_all_referenced_columns method which still says it does not support the  `having` clause

**Pull request merged**